### PR TITLE
Support GitHub Action

### PR DESCRIPTION
**Pull request by mrproliu**
* Support Github Action
* Fix exit code is not `1` bug when verify failed

### CODE REVIEW

File: action.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+runs:
+  using: "composite"
+  steps:
+    - shell: bash
+      run: |
+        mkdir -p skywalking-e2e/bin
+    - shell: bash
+      run: |
+        cd skywalking-e2e
+        git clone https://github.com/mrproliu/skywalking-infra-e2e.git
```
Comment (kezhenxu94) [inline action.yaml:34]:
```suggestion
        git clone https://github.com/apache/skywalking-infra-e2e.git
```

File: action.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        git -C skywalking-infra-e2e reset --hard 4fb6aef3f84b8d19bc38c75e4e4489d7ce1d8d71
```
Comment (kezhenxu94) [inline action.yaml:35]: Did you find any clue about how to get the revision if a user specify it in `uses: apache/skywalking-infra-e2e@main`, for example, get the `v1.0` in `uses: apache/skywalking-infra-e2e@v1.0`?

File: action.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        cd skywalking-e2e
+        git clone https://github.com/apache/skywalking-cli.git
+        git -C skywalking-cli reset --hard 21ba64dd2ddfd6cfeca14801e7c3888f1befc3ca
+        cd skywalking-cli
+        make build && mv bin/swctl-*-linux-amd64 ../bin/swctl
+    - shell: bash
+      run: |
+        cd skywalking-e2e/bin
+        pwd=`pwd`
+        export PATH="$PATH:$pwd"
```
Comment (kezhenxu94) [inline action.yaml:49]:
```suggestion
        export PATH="$PATH:$(pwd)"
```

File: test/verify/3.expected.yaml [hunk 4]
```diff
@@ -15,8 +15,10 @@
 # specific language governing permissions and limitations
 # under the License.
 
-{{- contains . }}
-- id: {{ notEmpty .id }}
-  name: {{ notEmpty .name }}
+{{if . }}
+- id: { { notEmpty .id } }
+  name: { { notEmpty .name } }
```
Comment (kezhenxu94) [inline test/verify/3.expected.yaml:20]: this is syntax error I believe

File: action.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  e2e-file:
+    description: File path of e2e file
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - shell: bash
+      run: |
+        mkdir -p skywalking-e2e/bin
```
Comment (kezhenxu94) [inline action.yaml:30]: Let's use an absolute path, for example, `/home/e2e`

**Review by wu-sheng — DISMISSED** [review]
Whoo, does this mean we have the first runnable version of this subproject?

**Review by kezhenxu94 — COMMENTED** [review]
Thanks, I left some comments, basically good to me

File: action.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
+#
```
Comment (wu-sheng) [inline action.yaml:1]: Should we run this task?

Comment (kezhenxu94) [reply action.yaml:1]: This is not a task, it's the definition of our GHA

File: action.yaml (hunk 5 above)
Comment (kezhenxu94) [reply action.yaml:30]: > Let's use an absolute path, for example, `/home/e2e`

Otherwise, it may pollute the users workspace

Comment (wu-sheng) [reply action.yaml:30]: please use `/home/skywalking-e2e`

File: action.yaml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  e2e-file:
+    description: File path of e2e file
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - shell: bash
+      run: |
+        mkdir -p /home/e2e/bin
```
Comment (kezhenxu94) [inline action.yaml:30]: see https://github.com/apache/skywalking-infra-e2e/pull/20#discussion_r654947942 please

File: action.yaml [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        git clone https://github.com/apache/skywalking-cli.git
+        git -C skywalking-cli reset --hard 21ba64dd2ddfd6cfeca14801e7c3888f1befc3ca
+        cd skywalking-cli
+        make build && mv bin/swctl-*-linux-amd64 ../bin/swctl
+    - shell: bash
+      run: |
+        pwd=`pwd`
+        export PATH="$PATH:/home/e2e/bin"
+        cd $pwd
+        e2e run -c "${{ inputs.e2e-file }}"
```
Comment (kezhenxu94) [inline action.yaml:50]:
```suggestion
        export PATH="$PATH:/home/e2e/bin"
        e2e run -c "${{ inputs.e2e-file }}"
```

File: action.yaml (hunk 5 above)
Comment (mrproliu) [reply action.yaml]: I have tried using an absolute path on this, but the GHA composite mode is not supported. please see: https://github.com/mrproliu/skywalking/pull/1/checks?check_run_id=2869437957

**Review by kezhenxu94 — APPROVED** [review]
https://github.com/apache/skywalking-infra-e2e/pull/20#discussion_r654945907 may not be an quick thing to do, so feel free to merge and consider it later, thanks very much for doing this 👍🏻

File: action.yaml (hunk 2 above)
Comment (wu-sheng) [reply action.yaml]: How SkyWalking eye works?

**Comment by mrproliu** [comment]
> [#20 (comment)](https://github.com/apache/skywalking-infra-e2e/pull/20#discussion_r654945907) may not be an quick thing to do, so feel free to merge and consider it later, thanks very much for doing this 👍🏻

Sorry for dismisses your approval, I need to do a little recheck. The https://github.com/apache/skywalking-infra-e2e/pull/20#discussion_r654945907 I will check if GHA has the variable after this PR merge.

File: action.yaml (hunk 2 above)
Comment (wu-sheng) [reply action.yaml]: GHA doesn't download the code base automatically?

Comment (mrproliu) [reply action.yaml]: > GHA doesn't download the code base automatically?

Using the `composite` type GHA, it doesn't download the code automatically.

Comment (kezhenxu94) [reply action.yaml]: There is a dilemma here, the SHA will be changed when we squash and merge, let me see

**Review by kezhenxu94 — APPROVED** [review]
Nice

**Pull request merged**